view_distillation/design_doc.md

```diff
@@ -0,0 +1,69 @@
+# View Distillation Design Doc
+
+## Input
+
+- A set of views
+  - For now, only allow views to be in csv format
+  - Can expand to different formats (ex: from databases) in the future
+
+## Create an instance
+
+`vd = ViewDistillation(path_to_views)`
+
+## APIs
+
+(The views are returned as paths.)
+
+### Find compatible views
+
+`result = vd.find_compatible_views()`
+
+This will return a list of lists, each inner list contains the views that are identical to each other.
+
+### Prune compatible views
+
+`result = vd.prune_compatible_views()`
+
+This will prune all compatible views and leave one, any future API calls will operate on the rest of views.
+It will also return the list of views left after pruning.
+
+### Find contained views
+
+`result = vd.find_compatible_views()`
+
+This will return a list of lists, each inner list contains the set of views in sorted order (descending by view size) 
```

Comment:
confusing use of list and set:

- First, why is the first list a list and not a set? 
- Second, the "inner list returns a set of views sorted" does not make sense. If it's sorted it is a list, not a set, right?